src/zoo_mcp/zoo_tools.py

```
@@ -445,6 +496,35 @@ async def zoo_convert_cad_file(
     return export_path
 
 
+async def zoo_execute_kcl(
+    kcl_code: str | None = None,
+    kcl_path: Path | str | None = None,
+) -> bool:
+    """Execute KCL code given a string of KCL code or a path to a KCL project. Either kcl_code or kcl_path must be provided. If kcl_path is provided, it should point to a .kcl file or a directory containing a main.kcl file.
+
+    Args:
+        kcl_code (str | None): KCL code
+        kcl_path (Path | str | None): KCL path, the path should point to a .kcl file or a directory containing a main.kcl file.
+
+    Returns:
+        bool: Returns True if the KCL code executed successfully, False otherwise.
+    """
+    logger.info("Executing KCL code")
+
+    kcl_code, kcl_path = _check_kcl_code_or_path(kcl_code, kcl_path)
+
+    try:
+        if kcl_code:
+            await kcl.execute_code(kcl_code)
+        else:
+            await kcl.execute(str(kcl_path))
+        logger.info("KCL code executed successfully")
+        return True
+    except Exception as e:
+        logger.info("Failed to execute KCL code: %s", e)
+        return False
```

Why not return the error message from `kcl.execute` if execution fails?

---

I can try to capture the error message, right now the `kcl.execute` method doesn't return the message only raises on failure.